packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -480,6 +481,669 @@ describe('postgresql - missing database', () => {
+                                                                                                                  `)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+  })
+
+  describe('postgresql views 23 - view with @@map', () => {
+    const warningCode = 23
+    setupPostgressForWarning(warningCode)
+
+    test('basic introspection', async () => {
+      ctx.fixture(`introspection/postgresql/views-warning-${warningCode}`)
+      const introspect = new DbPull()
+      const result = introspect.parse(['--print'])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["views"]
+        }
+
+        datasource db {
+          provider = "postgres"
+          url      = env("TEST_POSTGRES_URI_MIGRATE")
+        }
+
+        /// This view has been renamed to 'Renamedif' during introspection, because the original name 'if' is reserved.
+        /// The underlying view does not contain a valid unique identifier and can therefore currently not be handled by the Prisma Client.
+        view Renamedif {
+          id Int?
+
+          @@map("if")
+          @@ignore
+        }
+
+
+        // introspectionSchemaVersion: NonPrisma,
+      `)
+      expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.error'].mock.calls.join('\n')).toMatchInlineSnapshot(`
+
+                                                                                                                                                // *** WARNING ***
+                                                                                                                                                // 
+                                                                                                                                                // The following views were ignored as they do not have a valid unique identifier or id. This is currently not supported by the Prisma Client.
+                                                                                                                                                // - View "Renamedif"
+                                                                                                                                                // 
+                                                                                                            `)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+  })
+
+  describe('postgresql views 24 - no unique identifier', () => {
```

Comment:
TODO: Also test with multiple views to see that it does a list instead of 2 entries